toolchain/check/handle_where.cpp

```diff
@@ -67,12 +67,10 @@ auto HandleParseNode(Context& context, Parse::RequirementEqualId node_id)
     // If the type of the associated constant is symbolic, we defer conversion
     // until the constraint is resolved, in case it depends on `Self` (which
     // will now be a reference to `.Self`).
-    // TODO: It would be simpler to always defer this conversion until the
-    // constraint is resolved.
     // For now we convert to a value expression eagerly because otherwise we'll
     // often be unable to constant-evaluate the enclosing `where` expression.
-    // TODO: Find another way to handle this that allows us to only convert the
-    // RHS once.
+    // TODO: Perform the conversion symbolically and add an implicit constraint
+    // that this conversion exist.
```

Comment:
```suggestion
    // that this conversion exists.
```
(or "is valid" or something). I suppose we need to constrain not only that the conversion is possible but also that it produces a constant; might be worth mentioning that here too.

Reply:
Done.